File: MyEcommerce/MyEcommerce/Products/Category.cs
namespace MyEcommerce.Products
{
    public class Category
    {
        private int id;
        private string name;
        private List<Product> products;

        public Category(int id, string name)
        {
            this.id = id;
            this.name = name;
            products = new List<Product>();
        }


        public override string? ToString()
        {
            return this.name + " id = " + this.id;
        }

        public int Id { get { return id; } set { id = value; } }

        public List<Product> Products { get => products; set => products = value; }
    }
}

File: frontend/src/components/Category.js
import React from "react";
import CategoryStyle from "../style/CategoryStyle.module.css";
import { Link } from "react-router-dom";
import Product from "../style/ProductStyle.module.css";
import ArrowForwardRoundedIcon from "@mui/icons-material/ArrowForwardRounded";

function Category({ name, id }) {
  return (
    <div className={CategoryStyle.category}>
      <h4>{name}</h4>
      <Link to={`/categories/${id}/products`} className={Product.link}>
        <ArrowForwardRoundedIcon fontSize="large" />
      </Link>
    </div>
  );
}

export default Category;
